File: fix_home.py
import sys

file_path = "d:\\laravel20262\\arinifurniture\\resources\\views\\frontend\\home.blade.php"

with open(file_path, "r", encoding="utf-8") as f:
    content = f.read()

start_marker = "    <!-- HOME-SLIDER -->"
end_marker = "    <!-- //HOME-SLIDER -->"

if start_marker not in content or end_marker not in content:
    print("Markers not found!")
    sys.exit(1)

start_idx = content.find(start_marker)
end_idx = content.find(end_marker) + len(end_marker)

correct_slider = """    <!-- HOME-SLIDER -->
    <div class="home-slider yt-content-slider" data-autoplay="no" data-delay="4" data-speed="0.6" data-items_column00="1"
        data-margin="0" data-items_column0="1" data-items_column1="1" data-items_column2="1" data-items_column3="1"
        data-items_column4="1" data-arrows="yes" data-pagination="no" data-lazyload="yes" data-loop="yes"
        data-hoverpause="yes">
        @if (isset($sliders) && $sliders->count() > 0)
            @foreach ($sliders->chunk(2) as $sliderGroup)
                <div class="item-slider">
                    @foreach ($sliderGroup as $index => $slider)
                        <div class="item-slider-left {{ $index % 2 != 0 ? 'item-slider-right' : '' }}">
                            <a href="{{ $slider->link }}" class="">
                                <img src="{{ asset('storage/' . $slider->image) }}" alt="{{ $slider->title }}">
                            </a>
                            <div class="slider-info">
                                @if ($slider->subtitle)
                                    @if ($slider->type == 'slider_left')
                                        <h3 class="small-title">{{ $slider->subtitle }}</h3>
                                    @else
                                        <h2 class="big-title"><span class="small-title">{{ $slider->subtitle }}</span> {{ $slider->title }}</h2>
                                    @endif
                                @endif

                                @if ($slider->type == 'slider_left')
                                    <h4 class="small-desc">{{ $slider->description }}</h4>
                                    <h2 class="big-title">{{ $slider->title }}</h2>
                                @else
                                    @if (!$slider->subtitle)
                                        <h2 class="big-title">{{ $slider->title }}</h2>
                                    @endif
                                    <h4 class="small-desc">{{ $slider->description }}</h4>
                                @endif

                                <div class="text-center">
                                    <a href="{{ $slider->link }}" class="smooth see-more" title="">{{ $slider->button_text ?: 'SHOP NOW' }}</a>
                                </div>
                            </div>
                        </div>
                    @endforeach
                </div>
            @endforeach
        @else
            <!-- Fallback if no sliders in DB -->
            <div class="item-slider">
                <div class="item-slider-left">
                    <a href="#" class="">
                        <img src="{{ asset('frontend/image/catalog/demo/slider/home3/slider1.jpg') }}" alt="">
                    </a>
                    <div class="slider-info">
                        <h3 class="small-title">FURNITURE</h3>
                        <h4 class="small-desc">Survive The Reality Of Everyday Life</h4>
                        <h2 class="big-title">COLLECTION</h2>
                        <div class="text-center">
                            <a href="#" class="smooth see-more" title="">SHOP NOW</a>
                        </div>
                    </div>
                </div>
                <div class="item-slider-left item-slider-right">
                    <a href="#" class="">
                        <img src="{{ asset('frontend/image/catalog/demo/slider/home3/slider3.jpg') }}" alt="">
                    </a>
                    <div class="slider-info">
                        <h2 class="big-title"><span class="small-title">Locally made</span> sofas + sectionals</h2>
                        <h4 class="small-desc">Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod
                            tempor incididunt ut labore et dolore magna aliqua. </h4>
                        <div class="text-center">
                            <a href="#" class="smooth see-more" title="">SHOP NOW</a>
                        </div>
                    </div>
                </div>
            </div>
        @endif
    </div>
    <!-- //HOME-SLIDER -->

    <!-- BANNER -->
    <div class="banner" style="margin-top: 30px; margin-bottom: 30px;">
        <div class="container">
            <div class="row row0">
                @php
                    $fallbackImages = [
                        'frontend/image/catalog/demo/slider/home1/collection.png',
                        'frontend/image/catalog/demo/banners/home3/banner2.png',
                        'frontend/image/catalog/demo/banners/home3/banner3.png'
                    ];
                @endphp
                
                @for($i = 0; $i < 3; $i++)
                    <div class="col-lg-4 col-md-4 col-sm-4 col-xs-6 col0">
                        <div class="collection">
                            @if(isset($promoBanners) && isset($promoBanners[$i]))
                                <a href="{{ $promoBanners[$i]->link }}" class="hv-full-light">
                                    <img src="{{ asset('storage/' . $promoBanners[$i]->image) }}" class="img-responsive"
                                        alt="{{ $promoBanners[$i]->title }}">
                                </a>
                            @else
                                <a href="#" class="hv-full-light">
                                    <img src="{{ asset($fallbackImages[$i]) }}" class="img-responsive" alt="Fallback">
                                </a>
                            @endif
                        </div>
                    </div>
                @endfor
            </div>
        </div>
    </div>
    <!-- //BANNER -->"""

new_content = content[:start_idx] + correct_slider + content[end_idx:]

with open(file_path, "w", encoding="utf-8") as f:
    f.write(new_content)

print("Successfully fixed home.blade.php")

File: fix_home.php
<?php
$file = 'resources/views/frontend/home.blade.php';
$content = file_get_contents($file);

$startMarker = '<div class="featured-sale-box">';
$endMarker = '<!-- //CATEGORY-TAB-PRODUCT-SLIDER -->';

$startPos = strpos($content, $startMarker);
$endPos = strpos($content, $endMarker, $startPos);

if ($startPos !== false && $endPos !== false) {
    $endPos += strlen($endMarker);

    $replacement = <<<'EOD'
<div class="featured-sale-box">
    <h3 class="featured-sale-title">CATEGORIES <span>PRODUCTS</span></h3>
    <div id="so_listing_tabs_4" class="so-listing-tabs first-load">
        <div class="ltabs-wrap">
            <div class="ltabs-tabs-container clearfix" data-delay="300" data-duration="600"
                data-effect="starwars" data-ajaxurl="" data-type_source="0" data-lg="4"
                data-md="3" data-sm="2" data-xs="2" data-margin="30">
                <!--Begin Tabs-->
                <div class="ltabs-tabs-wrap">
                    <span class="ltabs-tab-selected"></span>
                    <span class="ltabs-tab-arrow">▼</span>
                    <div class="item-sub-cat">
                        <ul class="ltabs-tabs cf">
                            @foreach($categories as $index => $category)
                            <li class="ltabs-tab {{ $index == 0 ? 'tab-sel' : '' }}" data-category-id="{{ $category->id }}"
                                data-active-content=".category-{{ $category->id }}"> <span
                                    class="ltabs-tab-label smooth">{{ $category->name }}</span></li>
                            @endforeach
                        </ul>
                    </div>
                </div>
                <!-- End Tabs-->

                <div class="wap-listing-tabs ltabs-items-container products-list grid">
                    <!--Begin Items-->
                    @foreach($categories as $index => $category)
                    <div class="ltabs-items {{ $index == 0 ? 'ltabs-items-selected' : '' }} category-{{ $category->id }}" data-total="16">
                        <div class="ltabs-items-inner ltabs-slider ltabs00-4 ltabs01-3 ltabs02-3 ltabs03-2 ltabs04-1 none">
                            <div class="featured-tab-slider">
                                @foreach($products->where('category_id', $category->id) as $product)
                                <div class="item-slider">
                                    <div class="featured-product-box product-box clearfix">
                                        <div class="product-image">
                                            <a href="#" class="hv-radial">
                                                <img src="{{ $product->galleries->first() ? asset('storage/' . $product->galleries->first()->image) : asset('frontend/image/catalog/demo/products/product22.jpg') }}" class="img-responsive" alt="">
                                            </a>
                                        </div>
                                        <div class="product-info">
                                            <h4 class="product-name">
                                                <a href="#" class="smooth" title="">{{ $product->name }}</a>
                                            </h4>
                                            <div class="price">Rp {{ number_format($product->price, 0, ',', '.') }} </div>
                                            <div class="button-group">
                                                <button class="add-to-cart smooth" onclick="window.location.href='cart.html'">
                                                    ADD TO CART
                                                </button>
                                                <button class="wishlist-btn smooth" onclick="window.location.href='cart.html'">
                                                    <i class="fa fa-heart" aria-hidden="true"></i>
                                                </button>
                                            </div>
                                        </div>
                                    </div>
                                </div>
                                @endforeach
                            </div>
                        </div>
                    </div>
                    @endforeach
                    <!--End Items-->
                </div>
            </div>
        </div>
    </div>
<!-- //CATEGORY-TAB-PRODUCT-SLIDER -->
EOD;

    $newContent = substr($content, 0, $startPos) . $replacement . substr($content, $endPos);
    file_put_contents($file, $newContent);
    echo "Successfully replaced the categories products block.\n";
} else {
    echo "Could not find start or end markers.\n";
}
